reports/vamshi99.md

```diff
@@ -0,0 +1,188 @@
+   - details: >
+      GitLabUser: Fetch user id from username
+ - 16:
+   - repo: mr-rebaser
+   - link: https://gitlab.com/gitmate/open-source/mr-rebaser/commit/fad8305823a1058e93cd2397b5239a882b66c3f9
+   - details: >
+      Add squash pull request command
+ - 17:
+   - repo: mr-rebaser
+   - link: https://gitlab.com/gitmate/open-source/mr-rebaser/commit/3124c07ca5cd46acf985dc25fe22889e5a4744d0
+   - details: >
+      Dockerfile: Install requirements
+ - 18:
+   - repo: mr-rebaser
+   - link: https://gitlab.com/gitmate/open-source/mr-rebaser/commit/68360f3f0fac686907c7eb44c1a70bfa9e276d42
+   - details: >
+      run.py: Update squash command
+ - 19:
+   - repo: gitmate-2
+   - link: https://gitlab.com/gitmate/open-source/gitmate-2/commit/dda98a8ab67d01376d3199bb82260838cebab7b0
+   - details: >
+      rebaser: Add squash command
+ - 20:
+   - repo: IGitt
+   - link: https://gitlab.com/gitmate/open-source/IGitt/commit/f4ec3c602745bcf08405479a684876cc1080a21f
+   - details: >
+      Issue: Add available_assignees property
+ - 21:
+   - repo: gitmate-2
+   - link: https://gitlab.com/gitmate/open-source/gitmate-2/merge_requests/525
+   - details: >
+      Add assign request command
+ - 22:
+   - repo: gitmate-2
+   - link: https://gitlab.com/gitmate/open-source/gitmate-2/merge_requests/524
+   - details: >
+      Add review plugin
+---
+
+### Implement Gitmate automations and plugins for coala
+
+#### Work Done
+
+1. New Plugins added in GitMate:
+   1. Auto-assign issue based on commit mention in a Pull Request.
+   2. Notify merge conflicts in a Pull Request.
+   3. Auto review of Fixes vs closes keywords in a Pull request.
+   4. Auto reject pull requests opened on specified branches.
+   5. Squash commits in a Pull Request.
+   6. Request assignment to an issue.
+2. Add WIP label and remove pending review label if coala plugin finds defects
+   in a Pull Request.
+3. IGitt enhancements:
+   1. Filter issues by label, assignee in a repository.
+   2. Fetch user id from username in GitLabUser.
+   3. Added issues property and filter_issues method in Organization class.
+   4. Added fix_issues, close_issues, resolve_issues properties in
+      PullRequest.
+
+#### Challenges
+
+I had difficulties with implementing GitLab API in IGitt. GitHub has teams and
+GitLab API doesn't support teams, so I had to re-design assign request plugin
+to work without teams. Similarly, GitLab API doesn't have any webhook for pull
+request review event, so I couldn't implement a feature in GitMate
+which labels pull request as per review done by maintainer.
+
+#### Work to be done
+
+1. [Merge all pull request review plugins](https://gitlab.com/gitmate/open-source/gitmate-2/issues/242)
+   in GitMate into a single review plugin which will handle all pull request
+   states like work in progress, pending review, approved.
+2. [Mark review labels as per the review](https://gitlab.com/gitmate/open-source/gitmate-2/issues/296)
```

Comment:
Add more details into why it wasn't implemented. Being explicit is rather helpful here. One should at least have an overview here, rather than visiting the issue page.